seed/models/data_quality.py

```diff
@@ -235,6 +241,44 @@ class ComparisonError(Exception):
     pass
 
 
+def format_pint_violation(rule, source_value):
+    """
+    Format a pint min, max violation for human readability.
+
+    :param rule
+    :param source_value : Quantity - value to format into range
+    :return (formatted_value, formatted_min, formatted_max) : (String, String, String)
+    """
+
+    def pretty_units(q):
+        """
+        hack; can lose it when Pint gets something like a "{:~U}" format code
+        see https://github.com/hgrecco/pint/pull/231
+        """
+        return u"{:~P}".format(q).split(" ")[1]
+
+    formatted_min = formatted_max = None
+    incoming_data_units = source_value.units
+    rule_units = ureg(rule.units)
+    rule_value = source_value.to(rule_units)
+
+    pretty_source_units = pretty_units(source_value)
+    pretty_rule_units = pretty_units(rule_value)
+
+    if incoming_data_units != rule_units:
+        formatted_value = u"{:.1f} {} → {:.1f} {}".format(
+            source_value.magnitude, pretty_source_units,
+            rule_value.magnitude, pretty_rule_units,
+        )
+    else:
+        formatted_value = u"{:.1f} {}".format(source_value, pretty_rule_units)
```

Comment:
is this limiting the decimals to only the tenths place?

Reply:
Yeah. I tried out the `{:~P}` that Pint provides and was generally getting 10-15 decimal places and it was making the data quality message hard to grok. I don't know what the right number of decimal places is.

(`{:~.1P}`, annoyingly, would format the quantity in scientific, eg `1+e04 m²`).